src/main/java/com/comphenix/protocol/wrappers/WrappedLevelChunkData.java

```diff
@@ -0,0 +1,482 @@
+         * @return a mutable list containing {@link  BlockEntityInfo}
+         */
+        public List<BlockEntityInfo> getBlockEntityInfo() {
+            //noinspection StaticPseudoFunctionalStyleMethod
+            return Lists.transform((List<?>) BLOCK_ENTITIES_DATA_ACCESSOR.get(handle), BlockEntityInfo::new);
+        }
+
+        private static ConstructorAccessor levelChunkPacketDataConstructor;
+
+        /**
+         * Creates a new wrapper using predefined values.
+         *
+         * @param heightmapsTag   the heightmaps tag
+         * @param buffer          the buffer
+         * @param blockEntityInfo a list of wrapped block entities
+         * @return a newly created wrapper
+         */
+        public static ChunkData fromValues(NbtCompound heightmapsTag, byte[] buffer, List<BlockEntityInfo> blockEntityInfo) {
+            if (levelChunkPacketDataConstructor == null) {
+                levelChunkPacketDataConstructor = Accessors.getConstructorAccessor(HANDLE_TYPE, MinecraftReflection.getPacketDataSerializerClass(), int.class, int.class);
+            }
+
+            ConstructorAccessor trickySerializer = StructureCache.getTrickDataSerializerOrNull();
+
+            if (trickySerializer == null) {
+                throw new UnsupportedOperationException("TrickySerializer is not supported");
+            }
+
+            Object instance = levelChunkPacketDataConstructor.invoke(trickySerializer.invoke(new ZeroBuffer()), 0, 0);
+
+            BUFFER_ACCESSOR.set(instance, buffer);
+            HEIGHTMAPS_ACCESSOR.set(instance, NbtFactory.fromBase(heightmapsTag).getHandle());
+
+            for (BlockEntityInfo entityInfo : blockEntityInfo) {
+                //noinspection unchecked
+                ((List) BLOCK_ENTITIES_DATA_ACCESSOR.get(instance)).add(entityInfo.getHandle());
+            }
+
+            return new ChunkData(instance);
+        }
+    }
+
+    /**
+     * Wrapper for ClientboundLightUpdatePacketData
+     */
+    public static class LightData extends AbstractWrapper {
+
+        private static final Class<?> HANDLE_TYPE = MinecraftReflection.getLightUpdatePacketDataClass();
+
+        private static final List<FieldAccessor> BIT_SET_ACCESSORS;
+        private static final List<FieldAccessor> BYTE_ARRAY_LIST_ACCESSORS;
+
+        private static final FieldAccessor TRUST_EDGES_ACCESSOR;
+
+        static {
+            BIT_SET_ACCESSORS = asFieldAccessors(FuzzyReflection.fromClass(HANDLE_TYPE, true)
+                    .getFieldList(FuzzyFieldContract.newBuilder().typeExact(BitSet.class).build()));
+            BYTE_ARRAY_LIST_ACCESSORS = asFieldAccessors(FuzzyReflection.fromClass(HANDLE_TYPE, true)
+                    .getFieldList(FuzzyFieldContract.newBuilder().typeExact(List.class).build()));
+            TRUST_EDGES_ACCESSOR = Accessors.getFieldAccessor(FuzzyReflection.fromClass(HANDLE_TYPE, true)
+                    .getField(FuzzyFieldContract.newBuilder().typeExact(boolean.class).build()));
```

Comment:
Same here with caching fuzzy, and during init we can allow using streams once imo (asFieldAccessors -> stream().map()) :)